docs/DeveloperGuide.md

```diff
@@ -1,38 +1,1061 @@
+   recognised as a valid category and will not be processed correctly.
+   Example row: Apple Juice,5.50,Drink - In this case, the category is "Drink", which is not a valid category and
+   therefore not allowed.
+4. Not enough columns: If a row has insufficient columns compared to what is needed, the Entry cannot be created.
+   Example row: Apple Juice,5.50 - In this case, there are only two categories which is not allowed.
+
+<div style="text-align: right;">
+   <a href="#table-of-contents"> Back to Table of Contents </a>
+</div>
+
+# Product scope
+
+<div style="text-align: right;">
+   <a href="#table-of-contents"> Back to Table of Contents </a>
+</div>
+
+## Target user profile
 
 {Describe the target user profile}
 
-### Value proposition
+<div style="text-align: right;">
+   <a href="#table-of-contents"> Back to Table of Contents </a>
+</div>
+
+## Value proposition
 
 {Describe the value proposition: what problem does it solve?}
 
-## User Stories
+# User Stories
+
+| Version | As a ... | I want to ... | So that I can ... |
+| ------- | -------- | ------------- | ----------------- |
 
-|Version| As a ... | I want to ... | So that I can ...|
-|--------|----------|---------------|------------------|
-|v1.0|new user|see usage instructions|refer to them when I forget how to use the application|
-|v2.0|user|find a to-do item by name|locate a to-do without having to go through the entire list|
 
-## Non-Functional Requirements
+<div style="text-align: right;">
+   <a href="#table-of-contents"> Back to Table of Contents </a>
+</div>
+
+# Non-Functional Requirements
 
 {Give non-functional requirements}
 
-## Glossary
+<div style="text-align: right;">
+   <a href="#table-of-contents"> Back to Table of Contents </a>
+</div>
+
+<!-- # Glossary
 
 * *glossary item* - Definition
 
-## Instructions for manual testing
+<div style="text-align: right;">
+   <a href="#table-of-contents"> Back to Table of Contents </a>
+</div> -->
+
+# Acknowledgements
+
+## Documentation
+
+- [Github REST API documentation](https://docs.github.com/en/rest/quickstart?apiVersion=2022-11-28)
+- [Diagrams.net](https://app.diagrams.net/)
+- PlantUML
+
+## Storage
+
+- [Function `makeFileIfNotExists` - StackOverflow](https://stackoverflow.com/questions/9620683/java-fileoutputstream-create-file-if-not-exists)
+- [Deleting files - w3Schools](https://www.w3schools.com/java/java_files_delete.asp)
+- [BufferedReader - Baeldung](https://www.baeldung.com/java-buffered-reader)
+
+## Unit Tests
+
+- [Assert Exceptions Thrown - Baeldung](https://www.baeldung.com/junit-assert-exception)
+- [Arrange, Act, Assert](https://java-design-patterns.com/patterns/arrange-act-assert)
 
-{Give instructions on how to do a manual product testing e.g., how to load sample data to be used for testing}
+<div style="text-align: right;">
+   <a href="#table-of-contents"> Back to Table of Contents </a>
```

Comment:
The hyperlink at the end of the DG "A list of currently supported commands in PocketPal can be found [here]" to the user guide doesnt work